### PR TITLE
Hydrating Iterator

### PR DESCRIPTION
**Pull request by manchuck**
A hydrating iterator concept from @weierophinney's PhlyMongo module.  Basically will hydrate an object on each iteration.

### CODE REVIEW

File: src/Hydrator/Iterator/HydratingIteratorInterface.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Stdlib\Hydrator\Iterator;
+
+use \Iterator;
```
Comment (samsonasik) [inline src/Hydrator/Iterator/HydratingIteratorInterface.php]: backslash can be removed

File: src/Hydrator/Iterator/HydratingIteratorIterator.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Stdlib\Hydrator\Iterator;
+
+use \Iterator;
+use \IteratorIterator;
```
Comment (samsonasik) [inline src/Hydrator/Iterator/HydratingIteratorIterator.php]: backslash can be removed

**Comment by mwillbanks** [comment]
@weierophinney this one is for you :)

**Comment by weierophinney** [comment]
Merged to develop for release with 2.6.0.

**Pull request closed without merging**